Reactome base sonatype

jweiser opened the pull request:
This PR adds a directory ("reactome_base_sonatype") to the Curator Tool project which contains the additions necessary to build and deploy the "reactome-base" artifact to Sonatype for staging and then deployment to Maven Central.

The "lib" subfolder contains all the dependencies necessary for compiling the artifact.
The "pom.xml" file describes the artifact and its dependencies as required by Sonatype.
The "build.xml" file provides an ant script to build the artifact, its source jar and its JavaDocs jar and deploy them to Sonatype.

cookersjs commented on reactome_base_sonatype/build.xml:
```diff
@@ -0,0 +1,200 @@
+	<!-- define Maven coordinates -->
+	<property name="groupId" value="org.reactome.base" />
+	<property name="artifactId" value="reactome-base" />
+	<property name="version" value="1.2.22-SNAPSHOT" />
```
Should the snapshot naming convention be used?

jweiser replied:
Yes for this particular version because the first deployment to Sonatype was to the snapshot repository.  It will need to be changed if running the target to deploy to Sonatype as a release.

cookersjs commented on reactome_base_sonatype/pom.xml:
```diff
@@ -0,0 +1,218 @@
+        <connection>scm:git:git://github.com/reactome/CuratorTool.git</connection>
+        <developerConnection>scm:git:ssh://github.com/reactome/CuratorTool.git</developerConnection>
+        <url>https://github.com/reactome/CuratorTool</url>
+        <tag>v1.2.21</tag>
```
The tag version for curator tool is 1.2.21 here, but 1.2.22 for the version? I'm sure this wasn't missed, I'm just curious about the discrepancy

jweiser replied:
You are correct!  I didn't update the tag for the new version.  I will fix it - thanks!

jweiser replied:
Fixed!

SolomonShorser-OICR commented:
As per our conversation at your desk a minute ago: Remember to try using symlinks for binaries that are in more than one place. ;)

jweiser commented:
> As per our conversation at your desk a minute ago: Remember to try using symlinks for binaries that are in more than one place. ;)

Will do!  I'll try adding them and test if rebuilding works.

guanmingwu commented:
I believe all jar files have been listed in the original lib folder. In the new reactome_base_sonatype/lib, are these jar files copied over? If yes, would this cause inconsistence if a jar is updated in the lib folder? Would it be possible to use jar files in the original lib folder directly to avoid duplications?

jweiser commented:
> I believe all jar files have been listed in the original lib folder. In the new reactome_base_sonatype/lib, are these jar files copied over? If yes, would this cause inconsistence if a jar is updated in the lib folder? Would it be possible to use jar files in the original lib folder directly to avoid duplications?

Yes, you are right - I'm going to try making symbolic links to any duplicated library files and make a new commit.  Thanks!

guanmingwu commented:
Hi Joel,

Even not symbolic links. Would you ant script use jar files in the lib
folder directly? If not try copying on the fly then?

Thanks,

Guanming

On Sun, Sep 29, 2019 at 12:15 PM Joel Weiser <notifications@github.com>
wrote:

> I believe all jar files have been listed in the original lib folder. In
> the new reactome_base_sonatype/lib, are these jar files copied over? If
> yes, would this cause inconsistence if a jar is updated in the lib folder?
> Would it be possible to use jar files in the original lib folder directly
> to avoid duplications?
>
> Yes, you are right - I'm going to try making symbolic links to any
> duplicated library files and make a new commit. Thanks!
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/reactome/CuratorTool/pull/21?email_source=notifications&email_token=ABGIYOUTR6JGEU3Z6URW6E3QMD5EPA5CNFSM4I3J2DWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD734OQQ#issuecomment-536332098>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABGIYOQZM7HNVXLF25IW24LQMD5EPANCNFSM4I3J2DWA>
> .
>

jweiser commented:
> Hi Joel, Even not symbolic links. Would you ant script use jar files in the lib folder directly? If not try copying on the fly then? Thanks, Guanming
> […](#)
> On Sun, Sep 29, 2019 at 12:15 PM Joel Weiser ***@***.***> wrote: I believe all jar files have been listed in the original lib folder. In the new reactome_base_sonatype/lib, are these jar files copied over? If yes, would this cause inconsistence if a jar is updated in the lib folder? Would it be possible to use jar files in the original lib folder directly to avoid duplications? Yes, you are right - I'm going to try making symbolic links to any duplicated library files and make a new commit. Thanks! — You are receiving this because your review was requested. Reply to this email directly, view it on GitHub <#21?email_source=notifications&email_token=ABGIYOUTR6JGEU3Z6URW6E3QMD5EPA5CNFSM4I3J2DWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD734OQQ#issuecomment-536332098>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABGIYOQZM7HNVXLF25IW24LQMD5EPANCNFSM4I3J2DWA> .

Hi Guanming,

I have moved any library files which are not duplicates into the project base 'lib' directory and removed the 'reactome_base_sonatype/lib' directory.  I've also updated the build.xml to use the necessary library files in the project base 'lib' directory.

I also moved some of the files in the project base 'lib' directory into subdirectories if there were the same library files but of a different version.  The following libraries may now be obsolete and could possibly be deleted if it is okay with you since there are more recent or specific jar files now in the 'lib' directory:

- batik/batik-awt-util.jar
- batik/batik-dom.jar
- batik/batik-svggen.jar
- batik/batik-util.jar
- commons/commons-lang-2.3.jar
- jdom/jdom.jar
- jgraph/jgraph-5.7.4.6.jar
- log4j/log4j-1.2.12.jar
- ols-client/ols-client.jar

guanmingwu commented:
Would it be possible to make sure jar files used for the curator tool and the maven build be the same?

SolomonShorser-OICR commented on reactome_base_sonatype/pom.xml:
```diff
@@ -0,0 +1,218 @@
+
+    <licenses>
+        <license>
+            <name>GNU AFFERO GENERAL PUBLIC LICENSE</name>
```
Is this the license we're supposed to use for all of our projects?

SolomonShorser-OICR replied:
Or just this one?

jweiser replied:
Robin and I talked about it and decided to go with this license for this project.  I think this is what we would use going forward for all Reactome projects unless there was a specific reason to choose another license, but it is not mandatory.

SolomonShorser-OICR commented on reactome_base_sonatype/pom.xml:
```diff
@@ -0,0 +1,218 @@
+        <connection>scm:git:git://github.com/reactome/CuratorTool.git</connection>
+        <developerConnection>scm:git:ssh://github.com/reactome/CuratorTool.git</developerConnection>
+        <url>https://github.com/reactome/CuratorTool</url>
+        <tag>v1.2.22</tag>
```
Does that leading "v" get propagated to anything? The artefact? The repository tags?

jweiser replied:
It's specific to the GitHub repository tag, I believe.  I don't think it gets propagated to the jar file the POM file describes.

guanmingwu commented:
Joel, could you make sure the same versions of Jar files are used for the curator tool build and the reactome-based build?
